File: 10_laba/src/by/bntu/fitr/java/inGoslingsFootsteps/Lab10/model/logic/CustomerController.java
package by.bntu.fitr.java.inGoslingsFootsteps.Lab10.model.logic;

import by.bntu.fitr.java.inGoslingsFootsteps.Lab10.model.entity.AirTicketOfficeContext;
import by.bntu.fitr.java.inGoslingsFootsteps.Lab10.model.entity.Customer;
import by.bntu.fitr.java.inGoslingsFootsteps.Lab10.util.exception.InputException;
import org.apache.log4j.Logger;

import java.io.FileNotFoundException;

public class CustomerController {
    private static AirTicketOfficeContext context;
    private static final Logger LOG;
    static{
        LOG = Logger.getLogger(by.bntu.fitr.java.inGoslingsFootsteps.Lab10.model.logic.CustomerController.class);
    }

    public CustomerController() throws FileNotFoundException {
        context = AirTicketOfficeContext.getContext();
    }

    public Customer addNewCustomerOrGetExisting(String name, String email) throws InputException {
        if(email.equals("") || name.equals(""))
            throw new InputException("You entered incorrect data");
        Customer customer;
        for (int i = 0; i < context.sizeOfCustomers(); i++) {
            if (email.equals(context.getCustomerByIndex(i).getEmail())) {
                customer = context.getCustomerByIndex(i);
                LOG.info("Got existing customer");
                return customer;
            }
        }
        customer = new Customer(name, email);
        context.addCustomer(customer);
        LOG.info("Added new customer in json");
        return customer;
    }
}

File: 10_laba/C##/AirTicketOffice/AirTicketOffice/by/bntu/fitr/java/ingoslingsfootsteps/model/logic/CustomerController.cs
using AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.model.entity;
using AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.util.exception;
using NLog;
using System;
using System.Collections.Generic;
using System.Text;

namespace AirTicketOffice.by.bntu.fitr.java.ingoslingsfootsteps.model.logic
{
    public class CustomerController
    {
        private static readonly Logger logger = LogManager.GetCurrentClassLogger();

        private static AirTicketOfficeContext context;

        public CustomerController()
        {
            context = AirTicketOfficeContext.Context;
        }

        public Customer AddNewCustomerOrGetExisting(string name, string email)
        {
            if (name.Equals("") || email.Equals(""))
                throw new InputException("You entered the incorrect data");
            Customer customer;
            for (int i = 0; i < context.Customers.Length; i++)
            {
                if (email.Equals(context.GetCustomerByIndex(i).Email))
                {
                    customer = context.GetCustomerByIndex(i);
                    logger.Info("Got existing customer");
                    return customer;
                }
            }
            customer = new Customer(name, email);
            context.AddCustomer(customer);
            logger.Info("Added new customer in json");
            return customer;
        }
    }
}
